lib/game/base_game.dart

```diff
@@ -31,17 +32,15 @@ class BaseGame extends Game with FPSCounter {
   final List<Component> _removeLater = [];
 
   /// Current game viewport size, updated every resize via the [resize] method hook
-  Vector2 size;
+  final Vector2 size = Vector2.zero();
+  set size(Vector2 size) => this.size.setFrom(size);
 
   /// Camera position; every non-HUD component is translated so that the camera position is the top-left corner of the screen.
   Vector2 camera = Vector2.zero();
 
-  /// This method is called for every component added, both via [add] and [addLater] methods.
-  ///
-  /// You can use this to setup your mixins, pre-calculate stuff on every component, or anything you desire.
```

Comment:
I think this phrase is still useful in the new name of the method

Reply:
Ah true, I removed preAdd/prepare for a while completely until I realized the composed component needs it. 
I'll re-add the comments.